src/makeCalculator.js

```diff
@@ -4,7 +4,49 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: function (valueResult) {
+      return function (value) {
+        return valueResult + value;
+      };
+    },
+
+    subtract: function (valueResult) {
+      return function (value) {
+        return valueResult - value;
+      };
+    },
+
+    multiply: function (valueResult) {
+      return function (value) {
+        return valueResult * value;
+      };
+    },
+
+    divide: function (valueResult) {
+      return function (value) {
+        if (value > 0) {
+          return valueResult / value;
+        } else {
+          return 'Error';
+        }
+      };
+    },
+
+    operate: function (callback, value) {
+      this.result = callback(this.result)(value);
+
+      return this;
+    },
+
+    reset: function () {
+      this.result = 0;
+
+      return this;
+    },
+  };
```

Comment:
Use proper naming for the function as per the task description. The function is named `makeCalculator`, which is appropriate for the task.